src/components/ui/Breadcrumb/Breadcrumb.tsx

```diff
@@ -13,24 +20,86 @@ export interface BreadcrumbProps extends UIBreadcrumbProps {
   breadcrumbList: ItemElement[]
 }
 
-function Breadcrumb({ breadcrumbList }: BreadcrumbProps) {
+interface BaseBreadcrumbProps extends BreadcrumbProps {
+  isDesktop?: boolean
+}
+
+function BaseBreadcrumb({
+  breadcrumbList,
+  isDesktop = false,
+}: BaseBreadcrumbProps) {
+  const firstItem = isDesktop ? breadcrumbList[0] : null
+  const mediumItems = isDesktop
+    ? breadcrumbList.slice(1, -2)
+    : breadcrumbList.slice(0, -2)
+
+  const lastItems = breadcrumbList.slice(-2)
+
+  const collapseBreadcrumb = breadcrumbList.length > 4
+
   return (
-    <UIBreadcrumb divider="">
-      <Link aria-label="home" to="/">
+    <UIBreadcrumb
+      divider=""
+      className={isDesktop ? 'hidden-mobile' : 'display-mobile'}
+    >
+      <Link aria-label="Go to homepage" to="/">
         <Icon name="House" width={18} height={18} weight="bold" />
       </Link>
 
-      {breadcrumbList.map(({ item, name }, index) => {
-        return breadcrumbList.length === index + 1 ? (
-          <span key={String(index)}>{name}</span>
-        ) : (
-          <Link to={item} key={String(index)}>
-            {name}
-          </Link>
-        )
-      })}
+      {!collapseBreadcrumb &&
+        breadcrumbList.map(({ item, name }, index) => {
+          return breadcrumbList.length === index + 1 ? (
+            <span key={String(index)}>{name}</span>
+          ) : (
+            <Link to={item} key={String(index)}>
+              {name}
+            </Link>
+          )
+        })}
+
+      {collapseBreadcrumb && firstItem && (
+        <Link to={firstItem.item}>{firstItem.name}</Link>
+      )}
+
+      {collapseBreadcrumb && (
+        <UIDropdown>
+          <UIDropdownButton>
+            <span>...</span>
+          </UIDropdownButton>
+          <UIDropdownMenu data-store-breadcrumb-dropdown-menu>
+            {mediumItems.map(({ item, name }, index) => (
+              <UIDropdownItem
+                data-store-dropdown-menu-item
+                onClick={() => navigate(item)}
+                key={String(index)}
+              >
+                <Icon name="ArrowElbowDownRight" width={24} height={24} />
+                <span>{name}</span>
+              </UIDropdownItem>
+            ))}
+          </UIDropdownMenu>
+        </UIDropdown>
+      )}
+
+      {collapseBreadcrumb &&
+        lastItems.map(({ item, name }, index) => {
+          return lastItems.length === index + 1 ? (
+            <span key={String(index)}>{name}</span>
+          ) : (
+            <Link to={item} key={String(index)}>
+              {name}
+            </Link>
+          )
+        })}
     </UIBreadcrumb>
   )
 }
 
+const Breadcrumb = ({ breadcrumbList }: BreadcrumbProps) => (
+  <>
+    <BaseBreadcrumb breadcrumbList={breadcrumbList} />
+    <BaseBreadcrumb breadcrumbList={breadcrumbList} isDesktop />
```

Comment:
I think this will work to work around [that](https://github.com/vtex-sites/base.store/pull/436#discussion_r840737180) problem! Good job.